src/lib/syncLinkHrefWithContent.ts

```diff
@@ -0,0 +1,19 @@
+const MAILTO_PROTOCOL = "mailto:";
+
+/**
+ * Enforces contents of links to match their href.
+ * @param element HTMLAnchorElement or HTMLElement potentially container anchor elements
+ */
+export default function syncLinkHrefWithContent(element: HTMLElement) {
+  if (element.tagName === "A") {
+    const anchorElement = element as HTMLAnchorElement;
+    const url = new URL(anchorElement.href);
+    const prefix = url.protocol === MAILTO_PROTOCOL ? MAILTO_PROTOCOL : "";
+    anchorElement.href = prefix + anchorElement.textContent;
+    return;
+  }
+  const anchorElements = element.getElementsByTagName("a");
+  for (let i = 0; i < anchorElements.length; i++) {
```

Comment:
Why not:

```suggestion
  for (const elements of anchorElements) {
```

Reply:
It's slower.

Reply:
Another reason is... at the dom level you sometimes don't get normal arrays especially in older browsers and it gets really funky. It's safest to use it this way.